### PR TITLE
Added `Edge()` getter function

### PR DESCRIPTION
**Pull request by denk0403**
The `cgraph` package does not currently expose a `Graph` method for retrieving named edges without automatically creating one. So this PR simply adds a new method `Edge(name string, start *Node, end *Node)` with such functionality. Its implementation is heavily based on its node counterpart, [`Node(name string)`](https://github.com/goccy/go-graphviz/blob/865af036ddbb745c4424bbd2fdaa9a75668cf0d4/cgraph/cgraph.go#L916).

Additionally, in an effort to help improve this package's documentation, I added documentation only to the relevant methods.

### CODE REVIEW

File: cgraph/cgraph.go [hunk 1]
```diff
@@ -959,6 +964,9 @@ func (g *Graph) SubRep(n *Node) *SubNode {
 	}
 }
 
+// Creates a new edge with the given name in the graph, and returns the new edge.
+//
```
Comment (constraintAutomaton) [inline cgraph/cgraph.go:968]: It's a useless new line I think

**Comment by constraintAutomaton** [comment]
It's really cool, that documentation have been added for those methods, it's a bit annoying that most functions of that package have no documentation I'm gonna make an issue about it.

**Comment by goccy** [comment]
@denk0403 Thank you regarding the addition of documentation !

Regarding the `Edge` API, the `name` argument is fine, but I think it is difficult to use if `start` and `end` nodes are required. What is the actual use case ? (i.e., I am thinking that in cases where `start` and `end` can be specified, it may already be possible to retrieve the `edge` as well )

**Comment by denk0403** [comment]
> I am thinking that in cases where start and end can be specified, it may already be possible to retrieve the edge as well

Perhaps I missed it, but I didn't see an existing method for retrieving edges *without creating one if it doesn't exist*. So, a potential use case could be for querying if an edge of a known `name` exists between two nodes. It may be possible to implement this without requiring the `start` and `end` nodes, but my other goal was to continue to mirror the `C` API, which does require the nodes alongside the edge `name`.

**Comment by denk0403** [comment]
> I am thinking that in cases where `start` and `end` can be specified, it may already be possible to retrieve the `edge` as well

@goccy Perhaps I missed it, but I didn't see an existing method for retrieving edges *without creating one if it doesn't exist*. So, a potential use case could be for querying if an edge of a known `name` exists between two nodes. It may be possible to implement this without requiring the `start` and `end` nodes, but my other goal was to continue to mirror the `C` API, which does require the nodes alongside the edge `name`.

**Comment by goccy** [comment]
Thank you for your contribution !! I've already added `EdgeByName` and `EdgeByID` functions to get edge without creation.

**Pull request closed without merging**